src/oge/output_data.py

```diff
@@ -411,33 +385,29 @@ def write_generated_averages(
         skip_outputs (bool): whether to save data or not.
     """
     if not skip_outputs:
-        fuel_type_production = (
-            ba_fuel_data.groupby(["fuel_category"]).sum(numeric_only=True).reset_index()
+        national_avg = (
+            ba_fuel_data.groupby(["fuel_category"])[DATA_COLUMNS]
+            .sum(numeric_only=True)
+            .reset_index()
         )
 
         # Add row for total
-        total_production = fuel_type_production.sum(numeric_only=True).to_frame().T
-        total_production.loc[0, "fuel_category"] = "total"
-        production = pd.concat([fuel_type_production, total_production], axis=0)
-
-        # Calculate rates
-        for emission_type in ["_for_electricity", "_for_electricity_adjusted"]:
-            for emission in ["co2", "ch4", "n2o", "co2e", "nox", "so2"]:
-                production[f"generated_{emission}_rate_lb_per_mwh{emission_type}"] = (
-                    (
-                        production[f"{emission}_mass_lb{emission_type}"]
-                        / production["net_generation_mwh"]
-                    )
-                    .replace(np.inf, np.NaN)
-                    .replace(-np.inf, np.NaN)
-                    .fillna(0)
-                )
+        national_total = national_avg[DATA_COLUMNS].sum().reset_index()
```

Comment:
Should it be `ba_fuel_data.groupby(["fuel_category"])[DATA_COLUMNS].sum(numeric_only=True).reset_index()`?

Reply:
No we are calculating the "total" category so we want to just sum all of the existing data